kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/backstack/BackStackContainer.kt

```diff
@@ -72,9 +83,12 @@ open class BackStackContainer(
         }
 
     val newView = registry.buildView(named.top, this)
-    val popped = viewStateCache.update(named.backStack, oldViewMaybe, newView)
+    viewStateCache.update(named.backStack, oldViewMaybe, newView)
+
+    val popped = currentRendering?.backStack?.any { compatible(it, named.top) } == true
```

Comment:
Note in commit message that `BackStackContainer` no longer relies on its `ViewStateCache` to determine whether to run a push or pop effect. That choice is now completely deterministic, chosen by comparing the previous rendering with the new one.